core/src/executor/evaluate/function.rs

```diff
@@ -236,6 +236,20 @@ pub fn abs(name: String, n: Evaluated<'_>) -> Result<Value> {
     }
 }
 
+pub fn ifnull(expr1: Evaluated<'_>, then: Evaluated<'_>) -> Result<Value> {
+    // should we pass expr2 as an Expr enum and evaluate it if expr1 is null?
+    // that would be more efficient than evaluating it before hand..
+    // what do you think?
+
+    Ok(match expr1 {
+        Evaluated::Value(v) => match v.is_null() {
+            true => then.try_into()?,
+            false => v.into_owned(),
+        },
+        Evaluated::Literal(l) => l.try_into()?,
+    })
+}
```

Comment:
`Evaluated::Literal(l)` can be also `NULL`.
You can consider to use `expr.is_null()` to handle this.